packages/jellyfish-api-core/__tests__/category/spv/createAnchor.test.ts

```diff
@@ -0,0 +1,242 @@
+import { RpcApiError } from '@defichain/jellyfish-api-core'
+import { MasterNodeRegTestContainer, ContainerGroup, GenesisKeys } from '@defichain/testcontainers'
+import { ContainerAdapterClient } from '../../container_adapter_client'
+import BigNumber from 'bignumber.js'
+
+describe('Spv', () => {
+  const group = new ContainerGroup([
+    new MasterNodeRegTestContainer(GenesisKeys[0]),
+    new MasterNodeRegTestContainer(GenesisKeys[1]),
+    new MasterNodeRegTestContainer(GenesisKeys[2])
+  ])
+
+  const clients = [
+    new ContainerAdapterClient(group.get(0)),
+    new ContainerAdapterClient(group.get(1)),
+    new ContainerAdapterClient(group.get(2))
+  ]
+
+  beforeAll(async () => {
+    await group.start()
+
+    await setup()
+  })
+
+  afterAll(async () => {
+    await group.stop()
+  })
+
+  async function setMockTime (
+    clients: ContainerAdapterClient[], pastHour: number, futureHour = 0
+  ): Promise<void> {
+    const offset = Date.now() - (pastHour * 60 * 60 * 1000) + (futureHour * 60 * 60 * 1000)
+    for (let i = 0; i < clients.length; i += 1) {
+      await clients[i].misc.setMockTime(offset)
+    }
+  }
+
+  async function setup (): Promise<void> {
+    const anchorAuths = await group.get(0).call('spv_listanchorauths')
+    expect(anchorAuths.length).toStrictEqual(0)
+
+    // time travel back 13 hours ago
+    await setMockTime(clients, 13)
+
+    const anchorFreq = 15
+    for (let i = 0; i < anchorFreq; i += 1) {
+      const container = group.get(i % clients.length)
+      await container.generate(1)
+      await group.waitForSync()
+    }
+
+    {
+      const count = await group.get(0).getBlockCount()
+      expect(count).toStrictEqual(15)
+    }
+
+    // check the auth and confirm anchor mn teams at current height 15
+    await group.get(0).waitForAnchorTeams(clients.length)
+
+    // assertion for team
+    for (let i = 0; i < clients.length; i += 1) {
+      const container = group.get(i % clients.length)
+      const team = await container.call('getanchorteams')
+      expect(team.auth.length).toStrictEqual(clients.length)
+      expect(team.confirm.length).toStrictEqual(clients.length)
+      expect(team.auth.includes(GenesisKeys[0].operator.address))
+      expect(team.auth.includes(GenesisKeys[1].operator.address))
+      expect(team.auth.includes(GenesisKeys[2].operator.address))
+      expect(team.confirm.includes(GenesisKeys[0].operator.address))
+      expect(team.confirm.includes(GenesisKeys[1].operator.address))
+      expect(team.confirm.includes(GenesisKeys[2].operator.address))
+    }
+
+    {
+      // anchor auths length should be still zero
+      const auths = await group.get(0).call('spv_listanchorauths')
+      expect(auths.length).toStrictEqual(0)
+    }
+
+    // forward 3 hours from 12 hours ago and generate 15 blocks per hour
+    // set 3 hours because block height and hash chosen is then 3 hours
+    // then every 15 blocks will be matched again
```

Comment:
sorry did not understand this properly. could you explain it a bit more or point me to any reference?

Reply:
https://github.com/DeFiCh/ain/wiki/What-is-an-anchor%3F#anchor-data-creation